src/common/procedure/src/local.rs

```diff
@@ -429,7 +429,7 @@ impl ProcedureManager for LocalManager {
         logging::info!("LocalManager start to recover");
 
         let procedure_store = ProcedureStore::new(self.state_store.clone());
-        let messages = procedure_store.load_messages().await?;
+        let (messages, finished_ids) = procedure_store.load_messages().await?;
```

Comment:
Suggestion: Add some metrics for procedure recovery or record the cost time in the final log.